### ISSUE TITLE
Access Model on Service initialization (Question/Feature Request)

### ISSUE THREAD

**Issue by elennaro** [balderdashy/sails#3050]
I have a Service that should be initialized according to settings written in the database.
What is the correct way to achieve that goal?

**Comment by richdunajewski** [comment]
I use a startup hook which talks to the DB and grabs the configuration settings, then overrides the default configuration I have set in config/*.js and starts the server.

For example, our listen port is DB-derived, so at startup I check the DB for the port number the service should be configured at, and then tell Sails to use that port rather than the default of 1337 or 80.

If you mean the Sails services under api/services, then I use my startup hook to call an init() method in my service file, which gets run first. Then, each incoming request could call various methods within the service just like normal. This gives me a chance to do something once at startup but not pollute the config/bootstrap.js file.

Also, I find it helpful for my startup hook to delay returning control to Sails for about 2 seconds to ensure Grunt and anything else has a moment to complete before allowing Sails to begin listening for connections. I found that sometimes Sails would be so quick to accept connections that parallel tasks may still be finishing up and I had no way to wait for them to finish, so this was an acceptable workaround. In the case of Grunt copying a lot of assets on startup, clients would see a lot of 404s for assets when they refreshed after a restart, and this would make the UI look awful.

**Comment by elennaro** [comment]
Wow. Cool!

**Comment by sailsbot** [comment]
Thanks for posting, @elennaro.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
